tests/integration/init_file/runner.py

```diff
@@ -0,0 +1,54 @@
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+import time
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+PROJECT_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, "..", "..", ".."))
+SIGNAL_SIGTERM = 15
+INIT_FILE = os.path.join(SCRIPT_DIR, "auth.cypherl")
+
+
+def wait_for_server(port, delay=0.1):
+    cmd = ["nc", "-z", "-w", "1", "127.0.0.1", str(port)]
+    while subprocess.call(cmd) != 0:
+        time.sleep(0.01)
+    time.sleep(delay)
+
+
+def prepare_memgraph(memgraph_args):
+    memgraph = subprocess.Popen(list(map(str, memgraph_args)))
+    time.sleep(0.1)
+    assert memgraph.poll() is None, "Memgraph process died prematurely!"
+    wait_for_server(7687)
+    return memgraph
+
+
+def terminate_memgraph(memgraph):
+    pid = memgraph.pid
+    try:
+        os.kill(pid, SIGNAL_SIGTERM)
+    except os.OSError:
+        assert False, "Memgraph process didn't exit cleanly!"
+    time.sleep(1)
+
+
+def execute_test_restart_memgraph_with_init_file(memgraph_binary: str) -> None:
+    storage_directory = tempfile.TemporaryDirectory()
+    memgraph = prepare_memgraph([memgraph_binary, "--data-directory", storage_directory.name, "--init-file", INIT_FILE])
+    terminate_memgraph(memgraph)
+    memgraph = prepare_memgraph([memgraph_binary, "--data-directory", storage_directory.name, "--init-file", INIT_FILE])
+    terminate_memgraph(memgraph)
```

Comment:
Let's extend this test just a bit.
After the second memgraph is up, you can check that the user data is correct. 
Then you can kill that instance and start a new one using a clean directory and check that the user data is correct in that case as well.

Reply:
Yes agree, good comment :+1: